std/file.d

```diff
@@ -2297,44 +2319,55 @@ version(Posix) @safe unittest
     Throws:
         $(D FileException) on error.
   +/
-version(StdDdoc) string readLink(C)(const(C)[] link) @safe;
-else version(Posix) string readLink(C)(const(C)[] link) @safe
+version(StdDdoc) string readLink(R)(R link)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) ||
+        isConvertibleToString!R);
+else version(Posix) string readLink(R)(R link)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) ||
+        isConvertibleToString!R)
 {
-    static auto trustedReadlink(const(C)[] path, char[] buf) @trusted
+    static if (isConvertibleToString!R)
     {
-        return core.sys.posix.unistd.readlink(path.tempCString(), buf.ptr, buf.length);
+        return readLink!(convertToString!R)(link);
     }
-    static auto trustedAssumeUnique(ref C[] array) @trusted
+    else
     {
-        return assumeUnique(array);
-    }
+        alias posixReadlink = core.sys.posix.unistd.readlink;
+        enum bufferLen = 2048;
+        enum maxCodeUnits = 6;
+        char[bufferLen] buffer;
```

Comment:
I didn't write this, just indented it. Let's fix pre-existing issues separately.